hspec-discover/src/Test/Hspec/Discover/Sort.hs

```diff
@@ -0,0 +1,38 @@
+module Test.Hspec.Discover.Sort (
+  NaturalSortChunk(..)
+, naturalSortKey
+) where
+
+import           Data.Char
+
+data NaturalSortChunk
+  = NumericChunk { chunkNum :: Integer, chunkStr :: String }
+  | StringChunk { chunkStr :: String }
+  deriving (Eq, Show)
+
+instance Ord NaturalSortChunk where
+  compare (NumericChunk n1 s1) (NumericChunk n2 s2) =
+    compare n1 n2 ~> compareLength s1 s2 ~> compare s1 s2
+  compare (StringChunk s1) (StringChunk s2) = compare s1 s2
+  compare (NumericChunk _ _) (StringChunk _) = LT
+  compare (StringChunk _) (NumericChunk _ _) = GT
+
+(~>) :: Ordering -> Ordering -> Ordering
+EQ ~> a = a
+a ~> _ = a
+infixr 5 ~>
```

Comment:
- I think this is the same as `mappend` / `<>`.
- Even if we were to implement this ourself, I don't see the benefit of defining a operator for one-way use.  What I would want instead is a name that conveys what this function does.

Reply:
Oh I didn't realize there was a Monoid instance for Ordering. That's exactly what this is, great.